src/lib/converter/factories/declaration.ts

```diff
@@ -56,6 +56,10 @@ export function createDeclaration(context: Context, node: ts.Declaration, kind:
     let isExported: boolean;
     if (container.kindOf([ReflectionKind.Module, ReflectionKind.ExternalModule])) {
         isExported = false; // Don't inherit exported state in modules and namespaces
+    } else if (container.kindOf([ReflectionKind.TypeLiteral]) && context.converter.excludeNotExported) {
+        // If the container for this is a type literal, it never had the
+        // isExported flag set on it.  See https://github.com/TypeStrong/typedoc/issues/953
+        isExported = true;
```

Comment:
This worries me. Is there a reason we can't set the `isExported` flag on containers when creating them instead of this?

Reply:
Yeah, I looked into that.  That does seem like the more "correct" way to go about it. However, after extensive debugging I discovered that this factory function doesn't ever get called for object literals themselves.  Instead, I believe the associated Declaration is created here: https://github.com/TypeStrong/typedoc/blob/master/src/lib/converter/types/reference.ts#L127 

Perhaps there is an underlying issue where object literal declarations follow a different code path.  I didn't want to go there, so I just wrote some tests to determine the least invasive fix, which is what you see.

Happy to dive back in if you can provide some additional insights about code structure.  It's a pretty challenging codebase. :)

Reply:
Makes sense, thanks. I would like to fix this the right way, but haven't had enough time to dig in and see where that would be yet, so unfortunately I can't provide any guidance here. Maybe next weekend.

Yes, the codebase is rather difficult to follow, we're working on it, but its slow going. :)